### PR TITLE
Frontend header + components structure

### PR DESCRIPTION
**Pull request by jrgarou**
Base on [atom design patterns](http://demo.patternlab.io/) I tried to organize our components like this:

- `/foundations`: is the `/atom`. Inside it there's the `/base` with all our settings files(colors, sizes, themes, typography, animations...).
- `/widgets`: is the `/molecules`. MainHeaderLogo, MainNavigation, SearchInput...
- `/structures`: is the `/organisms`. Main, MainHeader, MainFooter...
- `/templates`: for now we have just MainTemplate only. But, with portal home page we could get some more
- `/pages`: Nothing inside yet

So, this PR is more `structure` than `header`, but all header assets is there.

**Feel free to change the names or restructure**

### CODE REVIEW

**Comment by eliawk** [comment]
Love your `atom design pattern` approach

File: src/styles/reset.scss [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  padding: 0;
+}
+
+textarea {
+  overflow: auto;
+  vertical-align: top;
+  resize: none;
+}
+
+// input:valid, textarea:valid
```
Comment (bent0b0x) [inline src/styles/reset.scss]: Can you remove these?

File: src/styles/reset.scss [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+button, input {
+  line-height: normal;
+}
+
+button, input[type="button"], input[type="reset"], input[type="submit"] {
+  cursor: pointer;
+  -webkit-appearance: button;
+  *overflow: visible;
+}
```
Comment (bent0b0x) [inline src/styles/reset.scss]: Need whitespace

File: src/styles/app.scss [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+  width: 100%;
 }
+
+//
+
+.full-block {
+  display: block;
+  height: 100%;
+  width: 100%;
+}
```
Comment (bent0b0x) [inline src/styles/app.scss]: Whitespace

File: src/scripts/components/widgets/MainHeaderLogo.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+`
+
+const MainLogoSVG = styled.svg`
+  display: block;
+  fill: ${props => props.theme.colors.header.logoFill};
+  height: 100%;
+  width: 100%;
+`
+
+//
```
Comment (bent0b0x) [inline src/scripts/components/widgets/MainHeaderLogo.js]: Looks like these `//` comments exist in many of the new components. Can you remove them?

File: src/scripts/components/templates/MainTemplate.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import Svgs from '../foundations/Svgs'
+
+//
```
Comment (bent0b0x) [inline src/scripts/components/templates/MainTemplate.js]: More `//`

File: src/styles/reset.scss (hunk 2 above)
Comment (jrgarou) [reply src/styles/reset.scss]: Feel free to remove all you want in there, @bent0b0x

**Pull request merged**